graph TD: panic "index out of range [100]" in createMapping when a chain reaches 25 nodes

A linear `graph TD` chain of 25 nodes crashes with a runtime panic; 24 nodes renders fine.

## Repro

```sh
{ echo "graph TD"; for i in $(seq 1 24); do echo "N$i --> N$((i+1))"; done; } | mermaid-ascii -f -
```

```
panic: runtime error: index out of range [100] with length 100

goroutine 1 [running]:
github.com/AlexanderGrooff/mermaid-ascii/cmd.(*graph).createMapping(0xc00003da70)
        .../cmd/graph.go:279 +0x12c5
github.com/AlexanderGrooff/mermaid-ascii/cmd.drawMap(0xc00022e150)
        ...
```

Replace `seq 1 24` with `seq 1 23` (24 nodes) and it renders correctly.

## Notes

- The threshold is exactly 25 nodes in a chain: 24 OK, 25 panics, and everything deeper (tested up to 200) panics the same way.
- `-x` / `-y` padding values do not move the threshold (tested `-y 1`, `-y 5`, `-y 10`), so the 100-cell grid indexed in `createMapping` appears to be in graph coordinates, with each chained node consuming 4 rows (25 × 4 = 100).
- Reproduced on Windows 11; nothing in the trace looks platform-specific.

Version: built from a4f2321 (`go install github.com/AlexanderGrooff/mermaid-ascii@latest` on 2026-07-17, module version v0.0.0-20260712165621-a4f23212201c) with Go 1.23.0.

Found while stress-testing mermaid-ascii against another terminal Mermaid renderer. This report was drafted with the help of an AI agent; the repro above was run and verified against a fresh build.

--- a/cmd/graph.go
+++ b/cmd/graph.go
@@ -181,13 +181,10 @@ func (g *graph) setSubgraphs(textSubgraphs []*textSubgraph) {
 }
 
 func (g *graph) createMapping() {
-	// Set mapping coord for every node in the graph
-	highestPositionPerLevel := []int{}
-	// Init array with 0 values
-	// TODO: I'm sure there's a better way of doing this
-	for i := 0; i < 100; i++ {
-		highestPositionPerLevel = append(highestPositionPerLevel, 0)
-	}
+	// Set mapping coord for every node in the graph.
+	// Keyed by level so it grows with the graph instead of assuming a fixed
+	// number of levels; a missing key reads as the zero value.
+	highestPositionPerLevel := map[int]int{}
 
 	// TODO: should the mapping be bottom-to-top instead of top-to-bottom?
 	// Set root nodes to level 0

--- a/cmd/render_graph_test.go
+++ b/cmd/render_graph_test.go
@@ -1,12 +1,32 @@
 package cmd
 
 import (
+	"fmt"
 	"strings"
 	"testing"
 
 	"github.com/AlexanderGrooff/mermaid-ascii/pkg/diagram"
 	"github.com/mattn/go-runewidth"
 )
+
+func TestRenderGraphHandlesLongChainWithoutPanic(t *testing.T) {
+	config := diagram.NewTestConfig(true, "cli")
+
+	var b strings.Builder
+	b.WriteString("graph TD\n")
+	for i := 1; i < 30; i++ {
+		fmt.Fprintf(&b, "N%d --> N%d\n", i, i+1)
+	}
+
+	output, err := RenderDiagram(b.String(), config)
+	if err != nil {
+		t.Fatalf("RenderDiagram() error = %v", err)
+	}
+
+	if !strings.Contains(output, "N30") {
+		t.Fatalf("expected output to contain the last node\noutput:\n%s", output)
+	}
+}
 
 func TestRenderGraphKeepsDisplayWidthForWideNodeLabels(t *testing.T) {
 	config := diagram.NewTestConfig(true, "cli")

